core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -518,7 +518,229 @@ void team_policy_check_valid_storage_level_argument(int level);
+      : internal_policy(space_, league_size_request, team_size_request,
+                        vector_length_request) {}
+
+  TeamPolicyCommon(const typename traits::execution_space& space_,
+                   int league_size_request, int team_size_request)
+      : internal_policy(space_, league_size_request, team_size_request, 1) {}
+
+  TeamPolicyCommon(const typename traits::execution_space& space_,
+                   int league_size_request, const Kokkos::AUTO_t&,
+                   int vector_length_request)
+      : internal_policy(space_, league_size_request, Kokkos::AUTO(),
+                        vector_length_request) {}
+
+  TeamPolicyCommon(const typename traits::execution_space& space_,
+                   int league_size_request, const Kokkos::AUTO_t&)
+      : internal_policy(space_, league_size_request, Kokkos::AUTO(), 1) {}
+
+  TeamPolicyCommon(const typename traits::execution_space& space_,
+                   int league_size_request, const Kokkos::AUTO_t&,
+                   const Kokkos::AUTO_t&)
+      : internal_policy(space_, league_size_request, Kokkos::AUTO(),
+                        Kokkos::AUTO()) {}
+  TeamPolicyCommon(const typename traits::execution_space& space_,
+                   int league_size_request, const int team_size_request,
+                   const Kokkos::AUTO_t&)
+      : internal_policy(space_, league_size_request, team_size_request,
+                        Kokkos::AUTO()) {}
+  /** \brief  Construct policy with the default instance of the execution space
+   */
+  TeamPolicyCommon(int league_size_request, int team_size_request,
+                   int vector_length_request)
+      : internal_policy(league_size_request, team_size_request,
+                        vector_length_request) {}
+
+  TeamPolicyCommon(int league_size_request, int team_size_request)
+      : internal_policy(league_size_request, team_size_request, 1) {}
+
+  TeamPolicyCommon(int league_size_request, const Kokkos::AUTO_t&,
+                   int vector_length_request)
+      : internal_policy(league_size_request, Kokkos::AUTO(),
+                        vector_length_request) {}
+
+  TeamPolicyCommon(int league_size_request, const Kokkos::AUTO_t&)
+      : internal_policy(league_size_request, Kokkos::AUTO(), 1) {}
+
+  TeamPolicyCommon(int league_size_request, const Kokkos::AUTO_t&,
+                   const Kokkos::AUTO_t&)
+      : internal_policy(league_size_request, Kokkos::AUTO(), Kokkos::AUTO()) {}
+  TeamPolicyCommon(int league_size_request, const int team_size_request,
+                   const Kokkos::AUTO_t&)
+      : internal_policy(league_size_request, team_size_request,
+                        Kokkos::AUTO()) {}
+
+  template <class... OtherProperties>
+  TeamPolicyCommon(const TeamPolicy<OtherProperties...> p)
+      : internal_policy(p) {
+    // Cannot call converting constructor in the member initializer list because
+    // it is not a direct base.
+    internal_policy::traits::operator=(p);
```

Comment:
Why exactly is it necessary to call the copy assignment operator here? Why isn't `internal_policy(p)` enough?
I see that this is just copied from the existing implementation and introduced in https://github.com/kokkos/kokkos/pull/3379.